components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -41,65 +49,100 @@
-
-			/* first, initialize velocity engine */
-			VelocityEngine velocityengine = new VelocityEngine();
+		VelocityEngine velocityengine = new VelocityEngine();
+		if (this.velocityLoggerName != null) {
+			velocityengine.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+									   "org.apache.velocity.runtime.log.Log4JLogChute");
+			velocityengine.setProperty("runtime.log.logsystem.log4j.logger", velocityLoggerName);
+		}
+		try {
 			velocityengine.init();
+		} catch (Exception e) {
+			log.error("Cannot initialize Velocity engine", e);
+			throw new APITemplateException("Cannot initialize Velocity engine", e);
+		}
+		VelocityContext context = configcontext.getContext();
 
-			Template t = velocityengine.getTemplate(this.getTemplatePath());
-
-			t.merge(context, writer);
+		return processTemplate(velocityengine, context, getVersionedWebAppTemplatePath());
+	}
 
+	@Override
+	public String getConfigStringForNonVersionedWebAppTemplate() throws APITemplateException {
+		VelocityEngine velocityengine = new VelocityEngine();
+		if (this.velocityLoggerName != null) {
+			velocityengine.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+									   "org.apache.velocity.runtime.log.Log4JLogChute");
+			velocityengine.setProperty("runtime.log.logsystem.log4j.logger", velocityLoggerName);
+		}
+		try {
+			velocityengine.init();
 		} catch (Exception e) {
-			log.error("Velocity Error", e);
-			throw new APITemplateException("Velocity Error", e);
+			log.error("Cannot initialize Velocity engine", e);
+			throw new APITemplateException("Cannot initialize Velocity engine", e);
 		}
-		return writer.toString();
+
+		ConfigContext configcontext = new APIConfigContext(this.api);
+		configcontext = new TransportConfigContext(configcontext, api);
+		configcontext = new ResourceConfigContext(configcontext, api);
+
+		VelocityContext context = configcontext.getContext();
+		context.put("apiContext", this.api.getContext());
+		String forwardAppContext = this.api.getContext();
+		if (forwardAppContext != null && forwardAppContext.charAt(0) == '/') {
+			forwardAppContext = forwardAppContext.substring(1);
+		}
+		context.put("forwardAppContext", forwardAppContext);
+		context.put("forwardAppVersion", this.api.getId().getVersion());
+
+		return processTemplate(velocityengine, context, getNonVersionedWebAppTemplatePath());
 	}
 
 	@Override
 	public OMElement getConfigXMLForTemplate(Environment environment) throws APITemplateException {
 		try {
-			return AXIOMUtil.stringToOM(getConfigStringForTemplate(environment));
+			return AXIOMUtil.stringToOM(getConfigStringForVersionedWebAppTemplate(environment));
 		} catch (XMLStreamException e) {
-			String msg =
-			             "Error converting string to OMElement - String: " +
-			                     getConfigStringForTemplate(environment);
+			String msg = "Error converting string to OMElement - String: " +
+					getConfigStringForVersionedWebAppTemplate(environment);
```

Comment:
fixed in commit eb9ee31a71178fc54788fbbf1b9ea11083fb6d32 of PR https://github.com/wso2/carbon-appmgt/pull/153